src/ModelManager.ts

```diff
@@ -93,90 +93,147 @@ export class ModelManager {
+                                        const data = this.modelStore.getData();
+                                        triggerPageModelLoaded(data);
+                                        return data;
+                                    });
+                                } else {
+                                    const data = this.modelStore.getData();
+                                    triggerPageModelLoaded(data);
+                                    return data;
+                                }
+                            } else if (!(this.modelClient.apiHost && (PathUtils.getCurrentURL() !== this.modelClient.apiHost))){
+                                throw new Error(`Attempting to retrieve model data from a non-browser.
+                                    Please provide the initial data with the property key model`
+                                );
+                            }
+                        });
+                    }
+                })
+            );
         }
+    }
 
+    /**
+     * Initializes the class fields for ModelManager
+     * If no path is provided, fallbacks are applied in the following order:
+     *
+     * @param {string|InitializationConfig} [config]                - configuration object
+     */
+    private _initializeFields(config?: ModelManagerConfiguration) {
         this._listenersMap = {};
         this._fetchPromises = {};
         this._initPromise = null;
 
+        this._modelClient = ((config && config.modelClient) || new ModelClient());
+
+        this._editorClient = new EditorClient(this);
+        this._clientlibUtil = new AuthoringUtils(this.modelClient.apiHost);
+    }
+
+    /**
+     * Returns paths required for fetching root model
+     * Order of preference for determining the root model url -
+     * 1. Page path provided via config
+     * 2. Meta property value for cq:pagemodel_root_url, if set
+     * 3. Model path contained in URL for default SPA.
+     * To add - For remote SPA opened within editor, path contained in parent i.e. AEM url
+     * 4. If none, it defaults to empty string
+     *
+     * @param {string|InitializationConfig} [config]                - configuration object
+     */
+    private _getPathsForModel(config?: ModelManagerConfiguration) {
+        // Model path explicitly provided by user in config
+        const path = config && config.path;
+
+        // Model path set statically via meta property
         const pageModelRoot = PathUtils.getMetaPropertyValue(MetaProperty.PAGE_MODEL_ROOT_URL);
         const metaPropertyModelUrl = PathUtils.internalize(pageModelRoot);
-        const currentPathname = PathUtils.getCurrentPathname();
-        const sanitizedCurrentPathname = currentPathname ? PathUtils.sanitize(currentPathname) : '';
+
+        const aemApiHost = this.modelClient.apiHost;
+        const isRemoteApp = aemApiHost && (PathUtils.getCurrentURL() !== aemApiHost);
```

Comment:
How does this work with SSR?

Reply:
This check is required only on browsers when fetching the pathname. I've updated the code accordingly. Thanks!